eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Configuration/ConfigResolver.php

```diff
@@ -301,39 +302,48 @@ private function logTooEarlyLoadedListIfNeeded($paramName)
         $resettableServiceIds = $container->getParameter('ezpublish.config_resolver.resettable_services');
         $updatableServices = $container->getParameter('ezpublish.config_resolver.updateable_services');
-        foreach (debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 10) as $t) {
+
+        // Lookup trace to find last service being loaded as possible blame for eager loading
+        // Abort if one of the earlier services is detected to be "safe", aka updatable
+        foreach (debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 35) as $t) {
             if (!isset($t['function']) || $t['function'] === 'getParameter' || $t['function'] === __FUNCTION__) {
                 continue;
             }
 
             // Extract service name from first service matching getXXService pattern
             if (\strpos($t['function'], 'get') === 0 && \strpos($t['function'], 'Service') === \strlen($t['function']) - 7) {
-                $serviceName = \strtolower(\preg_replace('/\B([A-Z])/', '_$1', \str_replace('_', '.', \substr($t['function'], 3, -7))));
+                $potentialClassName = \substr($t['function'], 3, -7);
+                $serviceName = \strtolower(\preg_replace('/\B([A-Z])/', '_$1', \str_replace('_', '.', $potentialClassName)));
 
-                // This (->setter('$dynamic_param$')) is safe as the system is able to update it on scope changes
+                // This (->setter('$dynamic_param$')) is safe as the system is able to update it on scope changes, abort
                 if (isset($updatableServices[$serviceName])) {
                     return;
                 }
 
-                // !! The remaining cases are not safe, typically:
+                // !! The remaining cases are most likely "not safe", typically:
                 // - ctor('$dynamic_param$') => this should be avoided, use setter or use config resolver instead
                 // - config resolver use in service factory => the service (or decorator, if any) should be marked lazy
 
-                // Exception are up-datable cases where we where unable to reverse engineer service name
-                // Which is the case for any class name based services, as namespace is omitted from compiled fn
+                // Possible exception: Class name based services, can't be resolved as namespace is omitted from
+                // compiled function. In this case we won't know if it was updateable and "safe", so we warn to be sure
                 if (!in_array($serviceName, $resettableServiceIds, true) && !$container->has($serviceName)) {
-                    // So in this case we are not sure if this is a warning or not, but we stay safe and warn about it
-                    // TODO: Might be possible to introspect the compiled container to extract class/service name
-                    $serviceName = '->' . $t['function'] . '()';
+                    $blame = $potentialClassName;
+                } else {
+                    $blame = '$' . $serviceName;
```

Comment:
What does `$<serviceName>` notation mean? This is rather confusing. If we need to hint that it is a service, we should use `@`. Not very strict as well, because it stands for a service injection, not a service, but at least a bit clearer. But I'm probably not following something here :)

Reply:
Indeed, typo here. Intention is to try to be as clear in output as possible what things are.
hence also the `$param$` output, even if the param use might not be dynamic param usage but direct config resolver usage where this is not the case.

```suggestion
                    $blame = '@' . $serviceName;
```